kairos-prover/methods/guest/src/main.rs

```diff
@@ -7,10 +7,14 @@ use risc0_zkvm::guest::env;
 
 risc0_zkvm::guest::entry!(main);
 
+use serde_json_wasm::to_vec;
+extern crate alloc;
+use alloc::vec::Vec;
+
 fn main() {
     let proof_inputs: ProofInputs = env::read();
 
-    let output = proof_inputs.run_batch_proof_logic().unwrap();
+    let output: Vec<u8> = to_vec(&proof_inputs.run_batch_proof_logic().unwrap()).unwrap();
 
-    env::commit(&output);
+    env::commit_slice(&output);
 }
```

Comment:
This change should be reverted.
Serialization in risc0 is very expensive going to JSON is not needed.
We are using risc0's sterilizer here already.

Reply:
@Avi-D-coder The issue is that the risc0 serialization is not supported in wasm. I need serde-json-wasm to deserialize the receipt in the contract runtime.

Reply:
Since we have agreed to use serde-json-wasm for the demo I wouldn't follow up on this issue for until we have a better means of serialization that we can use throughout the project (e.g. borsh).

Reply:
Update: Avi was right, we are working on this.

Reply:
Solved / Reverted => Issue closed.

Reply:
Unresolved because calling decode() on the journal throws a violation of gas rules exception in wasm:

```
"error_message": "Wasm preprocessing error: Enc
                                                            │ountered operation forbidden by gas rules. Consult instruct
                                                            │ion -> metering config map"
```

Reply:
Solved with rkyv now